react/server.js

```diff
@@ -0,0 +1,256 @@
+const app = express()
+const port = 3001
+dotenv.config()
+
+// Enviornment variables stored in the .env file you created
+const PELM_CLIENT_ID = process.env.PELM_CLIENT_ID
+const PELM_SECRET = process.env.PELM_SECRET
+const USER_ID = process.env.USER_ID
+
+
+app.use(
+  // Here we are using session in place of a database
+  // Instead use your own database to store values like the Authorization_Token
+  session({ secret: "secret", saveUninitialized: true, resave: true })
+);
+
+
+app.use(express.json())
+app.use(express.urlencoded({ extended: true }))
+
+
+// Create connect_token
+// Link to Pelm docs: https://docs.pelm.com/reference/post_auth-connect-token
+app.get('/connect', (req, res) => {
+
+  const headers = new Headers();
+  headers.set('Pelm-Client-Id', PELM_CLIENT_ID);
+  headers.set('Pelm-Secret', PELM_SECRET);
+
+  const encodedParams = new URLSearchParams();
+
+  encodedParams.set('user_id', USER_ID);
+
+  const requestOptions = {
+    method: 'POST',
+    headers,
+    body: encodedParams,
+  };
+
+  // Reach out to the Pelm API connect endpoint providing all the relevent headers
+  fetch('https://api.pelm.com/auth/connect-token', requestOptions)
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        return response.text().then(text => { throw new Error(text) })
+      }
+    })
+    .then((data) => {
+      // Return connect token
+      res.end(JSON.stringify({
+        connectToken: data['connect_token']
+      }))
+    })
+    .catch((error) => {
+      try {
+        const errorObject = JSON.parse(error.message);
+        console.log(errorObject)
+      } catch (e) {
+        console.log("an error occurred")
+      }
+    });
+})
+
+
+// Get energy_accouts
+// Link to Pelm docs: https://docs.pelm.com/reference/get_accounts
+app.get('/accounts', (req, res) => {
+
+  const accessToken = req.session.access_token
+  const headers = new Headers();
+  headers.set('Authorization', 'Bearer ' + accessToken);
+  headers.set('Pelm-Client-Id', PELM_CLIENT_ID);
+  headers.set('Pelm-Secret', PELM_SECRET);
+
+  const requestOptions = {
+    method: 'GET',
+    headers,
+  };
+
+  // Reach out to the Pelm API accounts endpoint providing all the relevent headers
+  fetch('https://api.pelm.com/accounts', requestOptions)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        return response.text().then((text) => {
+          throw new Error(text);
+        });
+      }
+    })
+    .then((data) => {
+      // Return accounts data
+      res.end(JSON.stringify(data))
+    })
+    .catch((error) => {
+      try {
+        const errorObject = JSON.parse(error.message);
+        console.log(errorObject);
+      } catch (e) {
+        console.log('an error occurred');
+      }
+    });
+})
+
+
+// Get energy usage intervals
+// Link to Pelm docs: https://docs.pelm.com/reference/get_intervals
+app.post('/intervals', (req, res) => {
```

Comment:
this should be `app.get`

Reply:
We need to use post since we are passing data from the front end. There are some situations where you can pass a body with get but express doesnt like that I believe

Reply:
gotcha, let's make all of these `app.post` then for consistency

Reply:
Sounds good